syntax/minisnip.vim

```diff
@@ -23,6 +23,18 @@ if exists("b:current_syntax")
     finish
 endif
 
+if !exists("g:minisnip_nested_syntax")
+    let g:minisnip_nested_syntax = 1
+endif
+
+if g:minisnip_nested_syntax
+    let filetype = get(matchlist(expand("%:t:r"), '_\([^_]\+\)_.\+$'), 1, '')
```

Comment:
I think this would be a better way of getting the filetype:
https://github.com/joereynolds/vim-minisnip/blob/master/autoload/minisnip.vim#L14

We should extract that out into a function i.e.
```
getFiletypeFromName(snippetName)
```

and use it where possible.

An issue I've spotted - snippets can have multiple filetypes associated with them. If a file is called `_javascript.ts_main` what's the expected syntax highlighting?

Also you'll need to update the docs under `*'g:minisnip_dir'*` as it mentions syntax there too (the last paragraph).

One final thing, if you can add some tests, you'll get brownie points. Sorry to be a pain, thanks for the PR :dancing_men:

Reply:
> I think this would be a better way of getting the filetype:
https://github.com/joereynolds/vim-minisnip/blob/master/autoload/minisnip.vim#L14

That is the inverse function, no? Generating a file name from a given filetype, whereas in the syntax file we are extracting the filetype from the filename.

> An issue I've spotted - snippets can have multiple filetypes associated with them. If a file is called _javascript.ts_main what's the expected syntax highlighting?

I would imagine in that case we would source syntax files from both javascript and ts. I do not use either language, so I can't speak to how successful that would be.

We could use the following:
```vim
    let filetype = get(matchlist(expand("%:t:r"), '_\([^_]\+\)_.\+$'), 1, '')
    if !empty(filetype)
        for ft in split(filetype, '\.')
            silent! exe "runtime! syntax/" . ft . ".vim"
            unlet! b:current_syntax
        endfor
    endif
```

Reply:
My mistake, you're about the filetype inferring.
I agree on your syntax idea, if there are multiple filetypes, use all of the filetypes for syntax highlighting